File: solutions/Java/[2042_Medium] Check if Numbers Are Ascending in a Sentence/areNumbersAscending.java
class Solution {
  public boolean areNumbersAscending(String s) {
    String[] parts = s.split(" ");
    int previous = -1;

    for (String p: parts) {
      if (p.charAt(0) >= '0' && p.charAt(0) <= '9') {
        int current = 0;

        for (int i = p.length() - 1, j = 1; i >= 0; i--, j *= 10) {
          current += (p.charAt(i) - '0') * j;
        }

        if (current <= previous) {
          return false;
        }

        previous = current;
      }
    }

    return true;
  }
}

File: solutions/TypeScript/[2042_Easy] Check if Numbers Are Ascending in a Sentence/areNumbersAscending.ts
function areNumbersAscending(s: string): boolean {
  const CHAR_CODE_0 = '0'.charCodeAt(0);
  const CHAR_CODE_9 = '9'.charCodeAt(0);

  const parts: string[] = s.split(' ');
  let previous = -1;

  for (const p of parts) {
    if (p.charCodeAt(0) >= CHAR_CODE_0 && p.charCodeAt(0) <= CHAR_CODE_9) {
      let current = 0;

      for (let i = p.length - 1, j = 1; i >= 0; i--, j *= 10) {
        current += (p.charCodeAt(i) - CHAR_CODE_0) * j;
      }

      if (current <= previous) {
        return false;
      }

      previous = current;
    }
  }

  return true;
}
